src/VisualStudio/IntegrationTest/New.IntegrationTests/AbstractEditorTest.cs

```diff
@@ -0,0 +1,92 @@
+// See the LICENSE file in the project root for more information.
+
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Shared.TestHooks;
+using Microsoft.VisualStudio.IntegrationTest.Utilities;
+using Roslyn.Test.Utilities;
+using Roslyn.Utilities;
+using Roslyn.VisualStudio.IntegrationTests.InProcess;
+
+namespace Roslyn.VisualStudio.IntegrationTests
+{
+    public abstract class AbstractEditorTest : AbstractIntegrationTest
+    {
+        private readonly string? _solutionName;
+        private readonly string? _projectTemplate;
+
+        protected AbstractEditorTest()
+        {
+        }
+
+        protected AbstractEditorTest(string solutionName)
+            : this(solutionName, WellKnownProjectTemplates.ClassLibrary)
+        {
+        }
+
+        protected AbstractEditorTest(string solutionName, string projectTemplate)
+        {
+            _solutionName = solutionName;
+            _projectTemplate = projectTemplate;
+        }
+
+        protected abstract string LanguageName { get; }
+
+        public override async Task InitializeAsync()
+        {
+            await base.InitializeAsync().ConfigureAwait(true);
+
+            if (_solutionName != null)
+            {
+                RoslynDebug.AssertNotNull(_projectTemplate);
+
+                await TestServices.SolutionExplorer.CreateSolutionAsync(_solutionName, HangMitigatingCancellationToken);
+                await TestServices.SolutionExplorer.AddProjectAsync(ProjectName, _projectTemplate, LanguageName, HangMitigatingCancellationToken);
+                await TestServices.SolutionExplorer.RestoreNuGetPackagesAsync(ProjectName, HangMitigatingCancellationToken);
+
+                // Winforms and XAML do not open text files on creation
+                // so these editor tasks will not work if that is the project template being used.
+                if (_projectTemplate is not WellKnownProjectTemplates.WinFormsApplication and
+                    not WellKnownProjectTemplates.WpfApplication and
+                    not WellKnownProjectTemplates.CSharpNetCoreClassLibrary and
+                    not WellKnownProjectTemplates.VisualBasicNetCoreClassLibrary)
+                {
+                    await TestServices.Editor.SetUseSuggestionModeAsync(false, HangMitigatingCancellationToken);
+                    await ClearEditorAsync(HangMitigatingCancellationToken);
+                }
+
+                // Work around potential hangs in 16.10p2 caused by the roslyn LSP server not completing initialization before solution closed.
+                // By waiting for the async operation tracking roslyn LSP server activation to complete we should never
+                // encounter the scenario where the solution closes while activation is incomplete.
+                await TestServices.Workspace.WaitForAsyncOperationsAsync(FeatureAttribute.LanguageServer, waitForWorkspaceFirst: true, HangMitigatingCancellationToken);
```

Comment:
@dibarbet do we still need this?

Reply:
Nope - should not be necessary

Reply:
Bug fix link https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1322125